worker/backend/cni_network.go

```diff
@@ -0,0 +1,242 @@
+package backend
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/go-cni"
+	"github.com/opencontainers/runtime-spec/specs-go"
+)
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 github.com/containerd/go-cni.CNI
+
+// CNINetworkConfig provides configuration for CNINetwork to override the
+// defaults.
+//
+type CNINetworkConfig struct {
+	// BridgeName is the name that the bridge set up in the current network
+	// namespace to connect the veth's to.
+	//
+	BridgeName string
+
+	// NetworkName is the virtual name used to identify the managed network.
+	//
+	NetworkName string
+
+	// Subnet is the subnet (in CIDR notation) which the veths should be
+	// added to.
+	//
+	Subnet string
+}
+
+// defaultCNINetworkConfig is the default configuration for the CNI network
+// created to put concourse containers into.
+//
+var defaultCNINetworkConfig = CNINetworkConfig{
+	BridgeName:  "concourse0",
+	NetworkName: "concourse",
+	Subnet:      "10.80.0.0/16",
+}
+
+const (
+	// fileStoreWorkDir is a default directory used for storing
+	// container-related files
+	//
+	fileStoreWorkDir = "/tmp"
```

Comment:
Do/should we ultimately point this at a subdirectory in the Concourse worker's `--work-dir`?